python/src/vmaf/core/asset.py

```diff
@@ -570,6 +574,23 @@ def dis_yuv_type(self):
         else:
             return self.DEFAULT_YUV_TYPE
 
+    @property
+    def desired_yuv_type(self):
+        """
+        for notyuv assets, we want to allow the decoded yuv format to be set by the user
+        this is highly relevant to image decoding, where we would like to select yuv444p
+        this property tries to read desired_yuv_type from asset_dict, if it is there it is set
+        else it default to default_yuv_type
+        """
+        allowed_yuv_types = [y for y in Asset.SUPPORTED_YUV_TYPES if y != 'notyuv']
```

Comment:
A better way is to convert to a set and exclude notyuv. See https://stackoverflow.com/questions/3462143/get-difference-between-two-lists

Reply:
+1 for workfile_yuv_type, I like it a lot!

not sure about supported_yuv_types: there is already .SUPPORTED_YUV_TYPES, but we want to exclude notyuv from it